pnet_datalink/src/lib.rs

```diff
@@ -211,11 +212,67 @@ impl NetworkInterface {
     pub fn mac_address(&self) -> MacAddr {
         self.mac.unwrap()
     }
-
+    
+    pub fn is_up(&self) -> bool {
+        self.flags & (pnet_sys::IFF_UP as u32) != 0
+    }
+    pub fn is_broadcast(&self) -> bool {
+        self.flags & (pnet_sys::IFF_BROADCAST as u32) != 0
+    }
     /// Is the interface a loopback interface?
     pub fn is_loopback(&self) -> bool {
         self.flags & (pnet_sys::IFF_LOOPBACK as u32) != 0
     }
+    pub fn is_point_to_point(&self) -> bool {
+        self.flags & (pnet_sys::IFF_POINTOPOINT as u32) != 0
+    }
+    pub fn is_multicast(&self) -> bool {
+        self.flags & (pnet_sys::IFF_MULTICAST as u32) != 0
+    }
+}
+
+impl ::std::fmt::Display for NetworkInterface {
+    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        let flags = if self.flags > 0 {
+            let rets = [
+                self.is_up(),
+                self.is_broadcast(),
+                self.is_loopback(), 
+                self.is_point_to_point(),
+                self.is_multicast()
+            ];
+            format!("{:X}<{}>", self.flags, 
+                rets.iter()
+                    .zip(FLAGS.iter())
```

Comment:
I find the design with the `FLAGS` global constant to be a bit fragile to changes. The order of the strings in that array must match the order of the function calls in the `rets` array here. And they are defined far away from each other. `FLAGS` is also not used anywhere else, so being a global constant does not serve the purpose of being available to lots of places.

Could you maybe move that global into this method instead?

Reply:
ok.